components/scream/scripts/gather_all_data.py

```diff
@@ -50,17 +51,20 @@ def formulate_command(self, machine):
 
         repo = scream_docs_repo if self._scream_docs else scream_repo
 
-        # Compute kokkos location
+        # Compute kokkos location. scream itself won't allow this, and test-all-scream will error out,
+        # but one can use gather-all-data for plenty of other apps, so we need to support this here.
+        local_cmd = self._run
         if self._kokkos:
-            kokkos_loc = pathlib.Path(self._kokkos.replace("$machine", machine).replace("$compiler", compiler))
+            # Note: kokkos_loc can use other magic strings, like $cxx_compiler, or $machine
+            kokkos_loc = self._kokkos
```

Comment:
If we don't want to support a custom kokkos for anything but scream-docs, you can put that expect check here.

Reply:
Well, I was thinking about it, but what if tomorrow someone decides to use gather-all for something else? I think gather-all might be able to accommodate that (with minor mods). test-all already errors out if you try to pass `-k blah` now, so you _would_ get an error.